{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Workshop: Assessment of Reusability\n",
    "\n",
    "Save your own local copy of this notebook by using File > Save a copy in Drive. At some point you may be prompted to trust the notebook. We promise that it is safe ðŸ¤ž\n",
    "\n",
    "**Disclaimer:**\n",
    "\n",
    "The Google Colab notebook environment will interpret any command as Python code by default. If we want to run shell commands, we will have to prefix them by `!`. So any command you see with a leading `!` is a shell command and if you wanted to run it in your terminal you would omit the leading `!`. For example, if in the Colab notebook you ran `!wget` you would just run `wget` in your terminal."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setup"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Launch the following cell to set up the appropriate environment in your Jupyter Notebook:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! pip install git+https://github.com/bokulich-lab/mishmash.git@lxml_change"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will download some data files and switch to working within the `materials` directory for the rest of the notebook."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! git clone https://github.com/lina-kim/nccr_winter_2026.git materials/\n",
    "% cd materials"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 1: Run a known dataset\n",
    "\n",
    "Input data files can be found in `data/`.\n",
    "\n",
    "ðŸ” **Question 1:** Open the data file you are assigned to and list the first three PMC IDs below:\n",
    "1. \n",
    "2. \n",
    "3. \n",
    "\n",
    "ðŸ” **Question 2:** Go to the article associated with each ID and record both the title and name of the first author below:\n",
    "1. \n",
    "2. \n",
    "3. \n",
    "\n",
    "ðŸ” **Question 3:** What kind of sequencing data (i.e. amplicon vs. shotgun metagenomic) do you expect to find with each paper? If you were to reuse these datasets, what other kinds of data would you need?\n",
    "1. \n",
    "2. \n",
    "3. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Complete the below code block for your assigned dataset and execute the MISHMASH command. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! mishmash assess_sequences \\\n",
    "    --pmc_input_file \\\n",
    "    --output_file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exercise 2: Evaluate a new research question\n",
    "\n",
    "Each group is given a category of publications to explore. \n",
    "1. Create a search query on [PubMed](https://pubmed.ncbi.nlm.nih.gov/). Save your search results as a CSV.\n",
    "2. Among your group members, split your search results by publication year.\n",
    "3. Use MISHMASH to assess the reproducibility of sequence data from the papers in your publication year.\n",
    "4. Compile your group's results and generate a figure to summarize your findings.\n",
    "5. At the end, present your figure and explain your data.\n",
    "\n",
    "Consider the following questions in your figure and brief presentations:\n",
    "* How many publications did you evaluate?\n",
    "* What trends do you see?\n",
    "* What hypotheses do you have that may explain these trends?\n",
    "* What are potential confounding factors in your interpretation?\n",
    "* If possible, generate a [PRISMA diagram](https://estech.shinyapps.io/prisma_flowdiagram/) to include in your slide."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Other options for search splits include:\n",
    "* In your search category, identify the 5-6 most represented journals. How many papers did you find per journal? -> Split your search results by journal and compare.\n",
    "* Consider microbiomes of different cancer types; each group member will take a specific type of cancer whose data may be available.\n",
    "* Consider the forensic microbiome; each group member will take a different body site whose data may be available.\n",
    "* Run `mishmash assess_metadata` and compare the MIxS schema used in different years of publications."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
